File: java/gov/nasa/gsfc/gmsec/api/mist/SubscriptionInfo.java
/*
 * Copyright 2007-2021 United States Government as represented by the
 * Administrator of The National Aeronautics and Space Administration.
 * No copyright is claimed in the United States under Title 17, U.S. Code.
 * All Rights Reserved.
 */


/**
 * @file SubscriptionInfo.java
 */

package gov.nasa.gsfc.gmsec.api.mist;

import gov.nasa.gsfc.gmsec.api.jni.mist.JNIConnMgrSubscriptionInfo;


/**
 * A class that contains the information the user has supplied when
 * setting up a subscription using any of the ConnectionManager subcribe() methods.
 *
 * @see ConnectionManager#subscribe(String)
 * @see ConnectionManager#subscribe(String, ConnectionManagerCallback)
 * @see ConnectionManager#unsubscribe(SubscriptionInfo)
 */
public class SubscriptionInfo
{
	private JNIConnMgrSubscriptionInfo m_jniSubInfo;


	// Prevent users from instantiating their own objects.
	private SubscriptionInfo()
	{
	}


	/** 
	 * This method is for internal GMSEC API use only.
	 * @param obj Object to reference for acquiring internal JNIConnMgrSubscriptionInfo.
	 * @return Internal JNIConnMgrSubscriptionInfo object.
	 */
	public static JNIConnMgrSubscriptionInfo getInternal(SubscriptionInfo obj)
	{
		return (obj == null ? null : obj.m_jniSubInfo);
	}


	/**
	 * This constructor is for internal GMSEC API use only.
	 * @param jSubInfo Internal JNIConnMgrSubscriptionInfo object.
	 */
	public SubscriptionInfo(JNIConnMgrSubscriptionInfo jSubInfo)
	{
		m_jniSubInfo = jSubInfo;
	}


	/**
	 * Returns the subject/topic string the user has supplied when setting
	 * up a subscription.
	 *
	 * @return Subject string associated with the SubscriptionInfo.
	 */
	public String getSubject()
	{
		return m_jniSubInfo.getSubject();
	}


	/**
	 * Returns the Callback object, if any, used when setting up the subscription.
	 *
	 * @return The ConnectionManagerCallback, if any, associated with the SubscriptionInfo.
	 */
	public ConnectionManagerCallback getCallback()
	{
		return m_jniSubInfo.getCallback();
	}
}

File: framework/src/gmsec4/mist/SubscriptionInfo.cpp
/*
 * Copyright 2007-2021 United States Government as represented by the
 * Administrator of The National Aeronautics and Space Administration.
 * No copyright is claimed in the United States under Title 17, U.S. Code.
 * All Rights Reserved.
 */


/** @file SubscriptionInfo.cpp
 *
 *  @brief This file contains the opaque wrapper for subscription info used
 *  when setting up and tearing down subscriptions to the middleware.
 */

#include <gmsec4/mist/SubscriptionInfo.h>

#include <gmsec4/mist/ConnectionManagerCallback.h>

#include <gmsec4/internal/mist/InternalSubscriptionInfo.h>

#include <gmsec4/SubscriptionInfo.h>



namespace gmsec
{
namespace api
{
namespace mist
{


SubscriptionInfo::SubscriptionInfo(gmsec::api::SubscriptionInfo* info, ConnectionManagerCallback* cb)
	: m_internal(new internal::InternalSubscriptionInfo(info, cb))
{
}


SubscriptionInfo::~SubscriptionInfo()
{
	delete m_internal;
}


gmsec::api::SubscriptionInfo* SubscriptionInfo::getInfo() const
{
	return m_internal->getInfo();
}


const char* SubscriptionInfo::getSubject() const
{
	return m_internal->getSubject();
}


ConnectionManagerCallback* SubscriptionInfo::getCallback() const
{
	return m_internal->getCallback();
}


void* SubscriptionInfo::getCallbackFunction() const
{
	return m_internal->getCallbackFunction();
}



void* SubscriptionInfo::getInternal() const
{
	return m_internal->getInternal();
}


void SubscriptionInfo::setInternal(void* internal)
{
	m_internal->setInternal(internal);
}


void SubscriptionInfo::setCallbackFunction(void* cb)
{
	m_internal->setCallbackFunction(cb);
}


} // namespace mist
} // namespace api
} // namespace gmsec
